beetsplug/edit.py

```diff
@@ -313,8 +352,9 @@ def apply_data(self, objs, old_data, new_data):
             if forbidden:
                 continue
 
-            id = int(old_dict['id'])
-            apply(obj_by_id[id], new_dict)
+            # Reconcile back the user edits, using the reference_field.
+            val = self.obj_from_ref(old_dict)
```

Comment:
> Here's one other crazy idea: give the objects a temporary id for the purpose of this workflow. Then, after the process is done, remove it before continuing with the import process.

That would certainly help lower the amount of complexity introduced on this pull request (taking care of https://github.com/beetbox/beets/pull/1846#discussion-diff-51828719 as well), and something that crossed my mind but felt like dangerous territory.

Are there any special considerations needed to be taken into account when assigning the temporary ids (ie. total_item_count + i)?